### ISSUE TITLE
Problemas al compartir en facebook

### ISSUE THREAD

**Issue by fitoat** [qtranslate/qtranslate-xt#842]
Buenas tardes

Tengo un sitio web en 4 idiomas, cuando voy a compartir en facebook la vista previa siempre me sale en español aunque el link sea a la nota en otro idioma.

Agradecería me indicaseis como solventar este problema.

Muchas gracias

**Comment by herrvigg** [comment]
It would be easier in english. If i understand right the sharing on FB does not take the language in the given URL. This has certainly something to do with the canonical URL because FB will crawl for it. There might be another plugin creating a conflict. Can't help much more with this.

**Comment by fitoat** [comment]
Thank you for you answer

**Comment by fitoat** [comment]
I have a website in 4 languages, when I go to share on Facebook the preview always comes out in Spanish even if the link is to the entry in another language.

**Comment by herrvigg** [comment]
Yes this is what i understood. I gave you a tip: check that your canonical URL are correct.
This github is not for general support, this is a free software and the investigations concerning your website are up to you :)
If you have a clear technical issue or feature you need in qTranslate please describe it properly. I will have to close this ticket otherwise.

**Comment by herrvigg** [comment]
Normally FB Open Graph uses this for its crawler:
`<meta property="og:url" content="https://example.com/path" />`

See : https://developers.facebook.com/docs/sharing/webmasters/getting-started/versioned-link/

But it has also to match the `<link rel="canonical" href="...">` otherwise the crawler will not work as expected.

**Comment by herrvigg** [comment]
-> to everyone here: how do you deal with the canonical links? The main one (link) and the one in Open Graph (meta)?

I've written my own handlers for all this but i don't really know the basic behavior with WordPress.

We have a `redirect_canonical` hook in qTranslate-XT. What does it do exactly? Shouldn't WordPress handle the normal link correctly?

For the Open Graph many third-party plugins can handle this but i don't know how well they work. I had some problems so i disabled everything and rewrote everything myself.
